src/log/JsonParsing.cc

```diff
@@ -50,6 +48,39 @@ void JsonParsing::addLogToFile(string key, string value) {
     log.close();
 }
 
+// adds a new log with current time as key and a given value
+// Format: "Day Hour:Minute:Second:Millisecond"
+void JsonParsing::addLogToFileTimestamp(string value) {
+
+    // read in the file
+    std::ifstream test(name_);
+    std::stringstream buffer;
+    buffer << test.rdbuf();
+
+    // make a json object out of the file contents
+    // and append a new key and value
+    json j;
+    if (!buffer.str().empty()) {
+      j = json::parse(buffer.str());
+    }
+
+    // calculate timestamp
+    // time since midnight, Janurary 1, 1970 UTC in milliseconds
+    unsigned long long timestamp = std::chrono::duration_cast<std::chrono::milliseconds>
+    (std::chrono::system_clock::now().time_since_epoch()).count();
+    
+    std::stringstream ss;
+    ss << timestamp;
+    // associate timestamp with given value
+    j[ss.str()] = { value };
+
+    // output the appended json object to the file
+    std::ofstream log;
+    log.open(name_);
+    log << j;
+    log.close();
+}
```

Comment:
The only thing I can think of is: the log entry would be operation type + timestamp + key + value. for example: "P+125234125+hello+world". 
When log entries are syncronized between nodes, it'd be a string instead of a key value pair. But when we store the logs in Json file, we store them in pair. so a real log stored in json file may be like: "125234125": "P+125234125+hello+world"
not sure am I right.
But if I'm right, we may need one more function which can provide a current timestamp to make up log entries.

Reply:
So we are trying to sync nodes with string not key-value?